clang/lib/Tooling/DependencyScanning/DependencyScanningCASFilesystem.cpp

```diff
@@ -140,20 +131,38 @@ void DependencyScanningCASFilesystem::scanForDirectives(
   using namespace llvm;
   using namespace llvm::cas;
 
+  auto reportCASError = [&](unsigned DiagID, Error Err) {
```

Comment:
What is going to happen if libclang issue a diagnostics? Can build system capture the diagnostics and display it correctly?

Or is it too much to make the change to propagate the errors all the way out and handle it on the top layer (either in clang-scan-deps or build system?)

Reply:
We return a `CXDiagnosticSet` from the libclang API when there are errors. There should already be a diagnostic consumer threaded through for this, so I don't think anything extra is needed here.

Reply:
If that is the case, then maybe we should enforce the Diagnostics in `reportCASError` otherwise abort? This `consumeError` will setup weird escapes.

Reply:
Good idea, it was only a unit test that didn't have a diagnostic engine already.